http-server-netty/src/main/java/io/micronaut/http/server/netty/RoutingInBoundHandler.java

```diff
@@ -639,57 +621,37 @@ private void handleStatusError(
             }
             filterAndEncodeResponse(
                     ctx,
-                    request,
                     nettyHttpRequest,
                     defaultResponse,
-                    MediaType.APPLICATION_JSON_TYPE,
-                    false
+                    MediaType.APPLICATION_JSON_TYPE
             );
         }
     }
 
     private void filterAndEncodeResponse(
             ChannelHandlerContext ctx,
-            HttpRequest<?> request,
-            NettyHttpRequest nettyHttpRequest,
+            NettyHttpRequest<?> request,
             MutableHttpResponse<?> finalResponse,
-            MediaType defaultResponseMediaType,
-            boolean skipOncePerRequest) {
+            MediaType defaultResponseMediaType) {
+
         AtomicReference<HttpRequest<?>> requestReference = new AtomicReference<>(request);
-        filterPublisher(
+        Flowable<MutableHttpResponse<?>> responsePublisher = Flowable.fromPublisher(filterPublisher(
```

Comment:
Please, use a custom subscriber instead of returning Flowables code

Reply:
Why? We would have to implement a custom `onErrorReturn`

Reply:
This change was made to improve performance by removing flowables flow

Reply:
I don't think we should be in the position of having to create a custom reactive library